Library/src/Systems/Users/UserSystem.cpp

```diff
@@ -177,17 +178,86 @@ void UserSystem::Login(const csp::common::String& UserName,
 	}
 }
 
-void UserSystem::RefreshSession(const csp::common::String& UserId, const csp::common::String& RefreshToken, LoginStateResultCallback Callback)
+void UserSystem::LoginWithRefreshToken(const csp::common::String& UserId, const csp::common::String& RefreshToken, LoginStateResultCallback Callback)
 {
-	auto Request = std::make_shared<chs_user::RefreshRequest>();
-	Request->SetDeviceId(csp::CSPFoundation::GetDeviceId());
-	Request->SetUserId(UserId);
-	Request->SetRefreshToken(RefreshToken);
+	if (CurrentLoginState.State == ELoginState::LoggedOut || CurrentLoginState.State == ELoginState::Error)
+	{
```

Comment:
Does this also need a `CurrentLoginState.State = ELoginState::LoginRequested;` at the start ?